ipa-core/src/net/test.rs

```diff
@@ -87,17 +120,18 @@ impl AddressableTestServer {
 /// Either a single Ring on MPC connection or all of the shards in a Helper.
 pub struct TestNetwork<F: ConnectionFlavor> {
     pub network: NetworkConfig<F>, // Contains Clients config
-    pub servers: Vec<AddressableTestServer>,
+    pub servers: Vec<Option<AddressableTestServer>>,
```

Comment:
is it what you want, or `Option<Vec<..>>` represents the state where you don't have ports assigned better?

Reply:
Not really, in here is more of a rust ownership constraint.
I need to be able to take the ports.

Reply:
this to me indicates that there is something off here. Consider consuming `self` and using `servers.into_iter()` to get `AddressableTestServer` instead of `&mut Option<AddressableTestServer>` that you can later replace

Reply:
Sounds good. This was a vestigial change before I used destructuring and zip in the `to_app` function. It was simple to remove. Good catch!